### PR TITLE
Source Github: add integration tests 

### LINKED ISSUE #34859: Source Github: add integration tests
### Topic

_No response_

### Relevant information

Add itnegration tests for empty streams

### PR DESCRIPTION
**Pull request by artem1205**
## What

Resolve #34859

## How

add integration tests

## Recommended reading order

1. `airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py`
2. `airbyte-integrations/connectors/source-github/unit_tests/integration/config.py`

## 🚨 User Impact 🚨

no breaking changes

## Pre-merge Actions
<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

### CODE REVIEW

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+from .config import ConfigBuilder
+
+_CONFIG = ConfigBuilder().with_repositories(["airbytehq/integration-test"]).build()
+
+
+def _create_catalog(sync_mode: SyncMode = SyncMode.full_refresh):
+    return CatalogBuilder().with_stream(name="events", sync_mode=sync_mode).build()
+
+
+class EventsTest(TestCase):
```
Comment (lazebnyi) [inline airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py:25]: Please add note to bypass reason that this stream tested with integration tests.

**Review by lazebnyi — APPROVED** [review]
LGTM. Need add note to bypass reason.

File: airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py (hunk 1 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py:25]: added

**Review by maxi297 — COMMENTED** [review]
I don't understand the need for a new lib when from a quick look, our internal framework can do as much. [Here](https://github.com/airbytehq/airbyte/pull/35069) is the first test migrate to the HttpMocker

The only thing I see different is the `OrderedRegistry` which I'm not sure why this is relevant for our test. If we have a case for that, we can add this to the integration tests framework we have

**Comment by artem1205** [comment]
@maxi297,
The only reason was about `headers` in `response` for pagination test. https://github.com/airbytehq/airbyte/issues/34910
After this change I can move to our internal framework.

**Comment by maxi297** [comment]
Can you check https://github.com/airbytehq/airbyte/pull/35105 @artem1205 ?

**Comment by artem1205** [comment]
@maxi297 , refactored, now our internal framework is used.

File: airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                query_params={"per_page": 100},
+            ),
+            HttpResponse(json.dumps(find_template("events", __file__)), 200),
+        )
+
+        source = SourceGithub()
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert len(actual_messages.records) == 2
+        assert all(("repository", "airbytehq/integration-test") in x.record.data.items() for x in actual_messages.records)
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py:92]: What behavior does this actually test? If this is a different behavior than `assert len(actual_messages.records) == 2` should we split this in another test and have descriptive test names that describe this?

File: airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+                body=json.dumps(find_template("events", __file__)),
+                status_code=200,
+            ),
+        )
+        source = SourceGithub()
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert len(actual_messages.records) == 4
+
+    def test_incremental_read(self):
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py:122]: Could we have a more descriptive name? Something like `test_given_state_more_recent_than_some_records_when_read_incrementally_then_filter_records`

File: airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        source = SourceGithub()
+        actual_messages = read(
+            source,
+            config=_CONFIG,
+            catalog=_create_catalog(sync_mode=SyncMode.incremental),
+            state=StateBuilder()
+            .with_stream_state("events", {"airbytehq/integration-test": {"created_at": "2022-06-09T10:00:00Z"}})
+            .build(),
+        )
+        assert len(actual_messages.records) == 1
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py:143]: It seems like there are other outcomes of incremental read that are probably interesting. Should we create other tests for `test_when_read_incrementally_then_emit_state_message` I don't see unit tests for Events so maybe we want to ensure the state value is correct too

File: airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100},
+            ),
+            HttpResponse('{"message":"some_error_message"}', 403),
+        )
+        source = SourceGithub()
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert len(actual_messages.records) == 0
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py:157]: I think a very interesting behavior when there are errors is the stream status that we end up with. The fact that there are records or not seems less relevant to me.

**Review by maxi297 — APPROVED** [review]
A couple of nits but I won't be blocking the review for that. Thanks for improving our safety net!

File: airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py (hunk 2 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py:107]: agreed, splitted into 2

File: airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py (hunk 3 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py]: renamed

File: airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py (hunk 5 above)
Comment (artem1205) [reply airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py]: added check for stream_status

**Pull request merged**